### ISSUE TITLE
Polymorphic serializer not working as expected for 400 Bad Request

### ISSUE THREAD

**Issue by mdefeche** [ghazi-git/drf-standardized-errors#11]
Hello,

First of all, thank you for the new release, it's very neat. I could totally get rid of my own custom AutoSchema =)

I noticed that the polymorphic serializer used for the 400 Bad Request is not rendered as expected on Redoc as shown below. I would expect to have `validation_error` and `parse_error` shown in the dropdown but it only shows `null`

While reading the code, I also noticed that the ParseErrorReponseSerializer is using the ClientErrorEnum class. Shouldn't it be the ParseErrorCodeEnum class instead ?

``` 
class ParseErrorResponseSerializer(serializers.Serializer):
    type = serializers.ChoiceField(choices=ClientErrorEnum.choices)
    errors = ParseErrorSerializer(many=True)
```

![image](https://user-images.githubusercontent.com/5990678/187271249-5ca9a416-f9bc-4018-9518-5c8fcd58d6f9.png)

Thanks again for the hard work !

**Comment by ghazi-git** [comment]
You're welcome and thanks for taking the time to report the issues you find.

About the parse errors, that's actually correct, the type should be `client_error` while the `code` is `parse_error` ([this page talks more about error types](https://drf-standardized-errors.readthedocs.io/en/latest/error_response.html#error-response-format)). So, a parse error response should look like
```
{
    "type": "client_error",
    "errors": [
        {
            "code": "parse_error",
            "detail": "Malformed request.",
            "attr": null
        }
    ]
}
```

To be honest, I use swagger UI. I assumed if it worked on swagger UI then it should work on redoc. But, I was wrong 🤦‍♂️  

Everything looks good on swagger UI (a "post /login" operation):
![image](https://user-images.githubusercontent.com/17159441/187284860-a98b6fd8-ad89-4054-9763-013a4ab50233.png)

But with redoc, I see the same issue as you (that weird null value as the first option whenever a polymorphic serializer is used)
![image](https://user-images.githubusercontent.com/17159441/187285330-cd15e802-7dcf-4f90-81bf-6837f33896f8.png)

I'll try to look more into it in the coming days and report back what I find out.

**Comment by mdefeche** [comment]
I'm using drf-spectacular PolymorphicSerializer in my projects and it's displayed correctly in Redoc
![image](https://user-images.githubusercontent.com/5990678/187524462-ad4d0a44-3739-4308-9b52-1a1c518f74ba.png)

Here is the corresponding OpenAPI element
```    
Question:
      oneOf:
      - $ref: '#/components/schemas/QuestionOption'
      - $ref: '#/components/schemas/QuestionOption'
      - $ref: '#/components/schemas/QuestionOption'
      - $ref: '#/components/schemas/QuestionTable'
      - $ref: '#/components/schemas/QuestionNothing'
      - $ref: '#/components/schemas/QuestionNothing'
      - $ref: '#/components/schemas/QuestionInfo'
      - $ref: '#/components/schemas/QuestionNothing'
      discriminator:
        propertyName: type
        mapping:
          CHOICE: '#/components/schemas/QuestionOption'
          MULTIPLE_CHOICES: '#/components/schemas/QuestionOption'
          ORDER: '#/components/schemas/QuestionOption'
          TABLE: '#/components/schemas/QuestionTable'
          INPUT: '#/components/schemas/QuestionNothing'
          DATE: '#/components/schemas/QuestionNothing'
          INFO: '#/components/schemas/QuestionInfo'
          RANGE: '#/components/schemas/QuestionNothing'
```
But for the 400 error body I get only one mapping (would expect two) and the type value is null (I guess both elements of the oneOf have a mapping to null, so one might override the other)
```
    DataClientsAccountsUpdateErrorResponse400:
      oneOf:
      - $ref: '#/components/schemas/DataClientsAccountsUpdateValidationError'
      - $ref: '#/components/schemas/ParseErrorResponse'
      discriminator:
        propertyName: type
        mapping:
          null: '#/components/schemas/ParseErrorResponse'
```

Thanks again

**Comment by ghazi-git** [comment]
The issue is fixed now (I just release a new version [0.12.1](https://github.com/ghazi-git/drf-standardized-errors/releases/tag/v0.12.1)), the schema generated should no longer have `null`, rather the correct mapping, sth like this:
```yaml
AuthLoginCreateErrorResponse400:
      oneOf:
      - $ref: '#/components/schemas/AuthLoginCreateValidationError'
      - $ref: '#/components/schemas/ParseErrorResponse'
      discriminator:
        propertyName: type
        mapping:
          validation_error: '#/components/schemas/AuthLoginCreateValidationError'
          client_error: '#/components/schemas/ParseErrorResponse'
```
The corresponding redoc page will show
![image](https://user-images.githubusercontent.com/17159441/188263967-5eae5bec-573c-445a-8c36-c26dfb21781b.png)

Before, I read in [polymorphic serializer docs](https://drf-spectacular.readthedocs.io/en/latest/drf_spectacular.html#drf_spectacular.utils.PolymorphicProxySerializer) that
> It is strongly recommended to pass the Serializers as list, and by that let drf-spectacular retrieve the field and handle the mapping automatically. In special circumstances, the field may not available when drf-spectacular processes the serializer. In those cases you can explicitly state the mapping with {'legal': LegalPersonSerializer, ...}, but it is then your responsibility to have a valid mapping.

So, I didn't question passing a list of serializers when everything looked fine in swagger UI. But, it seems like this is one of the special cases.

Anyway, thanks again for reporting the issue and all the details about it 🙏
